Basic Migrations command

XVincentX opened the pull request:
This pull request will introduce a migration mechanism for the gateway leveraging [node-migrate@next](https://npmjs.org/packages/migrate) module by Tj.

Connects #490
Connects #492 closes #492

Standing points:

- This module can track the ran migrations using a `.migrate` file or using a custom storage mechanism. Question is whether we want to use the file or store the things in Redis, for example.
If we keep them on Redis, you move the DB, you are ok. If you keep them on the file system you might re-run the migrations if you change the server. ✌️ 

- How do we make sure the users will run the migrations? I've checked Kong and they do not care. They rely on the fact that you read the change log and notice you've got a migration to run; therefore you run them. I'd stick with the same mechanism ✌️ 

- [x] Figure out a way to test the effects by this migration
~Integrate the migration command with the `eg` cmd line~ (we're not doing that)

XVincentX commented:
// @kevinswiber @DrMegavolt What's your idea around the standing points?

DrMegavolt commented:
what is wrong with just running `migrate` ? why do we need `eg` cli integration ?

XVincentX commented:
I thought it was a requirement in order to give to the user an uniform interface to use the gateway.
Anyway, `npx migrate` or execute the bin file directly will do the job perfectly and it'll also save me some work. I'm fine removing this 👍

XVincentX commented:
I removed the CLI integration and we'll stick with the default storing mechanism! 👍

XVincentX commented:
@DrMegavolt Please review this one ore time. I've modified the code to use the `dao` instead of the `service` — this is to skip the password hashing and have a simple password copy. In this way our users won't have to communicate any new password.

Also, I've added a test that should check the whole flow.

DrMegavolt commented on test/migrations/credential-username-userid-transform.js:
```diff
@@ -0,0 +1,55 @@
+    let tmpFile;
+    let userId;
+    let oldCredential;
+    before('I insert some credentials by user ID', () => {
```
I guess the name should be 
`I insert some credentials by username`

XVincentX replied:
Yeah!
